mpp/shmem_c_func.m4

```diff
@@ -5,105 +5,114 @@ dnl This file is part of the Sandia OpenSHMEM software package. For license
 dnl information, see the LICENSE file in the top level directory of the
 dnl distribution.
 dnl
+
+#ifndef SHMEM_FUNCTION_ATTRIBUTES
+#  if OPAL_C_HAVE_VISIBILITY == 1
+#     define SHMEM_FUNCTION_ATTRIBUTES __attribute__((visibility("default")))
+#  else
+#     define SHMEM_FUNCTION_ATTRIBUTES
+#  endif
+#endif
```

Comment:
This was actually the simplest way I could think of to hide all unused symbols even with --enable-profiling.  If you remove this block then nothing links with profiling on.  It seems like #ifdef ENABLE_PROFILING here won't work, so I'm open to other ideas.  If I understand correctly, [OMPI does this as well](https://github.com/open-mpi/ompi/blob/master/ompi/include/mpi.h.in#L209).